packages/dev/core/src/Shaders/ShadersInclude/pbrBRDFFunctions.fx

```diff
@@ -138,11 +138,114 @@ float fresnelSchlickGGX(float VdotH, float reflectance0, float reflectance90)
         #else
             vec3 s = sqrt(f0);
             vec3 t = (vClearCoatRefractionParams.z + vClearCoatRefractionParams.w * s) / (vClearCoatRefractionParams.w + vClearCoatRefractionParams.z * s);
-            return t * t;
+            return square(t);
         #endif
     }
 #endif
 
+#ifdef IRIDESCENCE
+// XYZ to sRGB color space
+const mat3 XYZ_TO_REC709 = mat3(
+     3.2404542, -0.9692660,  0.0556434,
+    -1.5371385,  1.8760108, -0.2040259,
+    -0.4985314,  0.0415560,  1.0572252
+);
+
+// Assume air interface for top
+// Note: We don't handle the case fresnel0 == 1
+vec3 getIORTfromAirToSurfaceR0(vec3 f0) {
+    vec3 sqrtF0 = sqrt(f0);
+    return (1. + sqrtF0) / (1. - sqrtF0);
+}
+
+// Conversion FO/IOR
+vec3 getR0fromIORs(vec3 iorT, float iorI) {
+    return square((iorT - vec3(iorI)) / (iorT + vec3(iorI)));
+}
+
+float getR0fromIORs(float iorT, float iorI) {
+    return square((iorT - iorI) / (iorT + iorI));
+}
+
+// Fresnel equations for dielectric/dielectric interfaces.
+// Ref: https://belcour.github.io/blog/research/2017/05/01/brdf-thin-film.html

```

Comment:
This url does not work for me (404). However, this one is ok:

https://belcour.github.io/blog/research/publication/2017/05/01/brdf-thin-film.html